core-java-modules/core-java-jndi/src/test/java/com/baeldung/jndi/exceptions/JndiExceptionsUnitTest.java

```diff
@@ -0,0 +1,36 @@
+package com.baeldung.jndi.exceptions;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.jndi.JndiTemplate;
+import org.springframework.mock.jndi.SimpleNamingContextBuilder;
+
+import javax.naming.InitialContext;
+import javax.naming.NameNotFoundException;
+import javax.naming.NoInitialContextException;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class JndiExceptionsUnitTest {
+
+    @Test
+    public void testNoInitialContextException() {
```

Comment:
The test names in this class need updating to given/when/then - try to keep the names a little more terse.